src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/ValidationRules/MaxNumberOfResultsValidationRule.cs

```diff
@@ -0,0 +1,57 @@
+using GraphQL.Language.AST;
+using GraphQL.Validation;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+
+namespace OrchardCore.Apis.GraphQL.ValidationRules
+{
+    public class MaxNumberOfResultsValidationRule : IValidationRule
+    {
+        private readonly int _maxResults;
+        private readonly MaxNumberOfResultsValidationMode _maxNumberOfResultsValidationMode;
+
+        public MaxNumberOfResultsValidationRule(int maxNumberOfResults, MaxNumberOfResultsValidationMode maxNumberOfResultsValidationMode)
+        {
+            _maxResults = maxNumberOfResults;
+            _maxNumberOfResultsValidationMode = maxNumberOfResultsValidationMode;
+        }
+
+        public INodeVisitor Validate(ValidationContext validationContext)
+        {
+            return new EnterLeaveListener(_ =>
+            {
+                _.Match<Argument>(arg =>
+                {
+                    if ((arg.Name == "first" || arg.Name == "last") && arg.Value != null)
+                    {
+                        var context = (GraphQLContext)validationContext.UserContext;
+
+                        var value = (IntValue)arg.Value;
+
+                        if (value?.Value > _maxResults)
+                        {
+                            var localizer = context.ServiceProvider.GetService<IStringLocalizer<MaxNumberOfResultsValidationRule>>();
+                            var errorMessage = localizer[$"'{value}' exceeds the maximum number of results for '{arg.Name}' ({_maxResults})"];
```

Comment:
You can't use dynamic string in localization.

```suggestion
                            var errorMessage = localizer["'{0}' exceeds the maximum number of results for '{1}' ({2})", value, arg.Name, _maxResults];
```

Reply:
```csharp
var errorMessage = localizer["'{0}' exceeds the maximum number of results for '{1}' ({2})", value, args.Name, _macResults];
```

Reply:
@hishamco what are you tying to say?

Reply:
The fix coz I already did such mistake in another PR :)

Reply:
Ok, I thought you had just duplicated my line, and introduced a bug. Or maybe you didn't that my comment contained an actionable GH suggestion too.

Reply:
This is happen on mobile and slow internet connection, you edit while someone else edit the same thing 😂